#!/usr/bin/env python3

PACKAGE = "shifted_line_pkg"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# param, data_type, 0, description, defaut, min, max
gen.add("canny_low", double_t, 0, "hue low", 60, 0, 255)
gen.add("canny_high", double_t, 0, "hue high", 180, 0, 255)

gen.add("hue_l", double_t, 0, "hue low", 10, 0, 255)
gen.add("hue_h", double_t, 0, "hue high", 40, 0, 255)

gen.add("sat_l", double_t, 0, "sat low", 0, 0, 255)
gen.add("sat_h", double_t, 0, "sat high", 255, 0, 255)

gen.add("val_l", double_t, 0, "val low", 100, 0, 255)
gen.add("val_h", double_t, 0, "val high", 255, 0, 255)

gen.add("rho", double_t, 0, "resolution of the line in pixels", 0.8, 0.1, 1.0)
gen.add("theta", int_t, 0, "angle of the line in degrees", 45, 1, 180)
gen.add("thresh", int_t, 0, "minimum number of votes (intersections)", 15, 0, 100)
gen.add("min_length", int_t, 0, "minimal number of pixels making up a line", 200, 0, 600)
gen.add("max_gap", int_t, 0, "maximum line gap between connectable line segments in pixels", 50, 0, 200)


# the third parameter must match this file's name
exit(gen.generate(PACKAGE, "detect_yellow", "DetectYellow"))
